src/pkg/backup/details/details_test.go

```diff
@@ -469,6 +484,64 @@ func (suite *DetailsUnitSuite) TestDetails_Add_ShortRefs() {
 	}
 }
 
+func (suite *DetailsUnitSuite) TestDetails_Add_ShortRefs_Unique_From_Folder() {
+	t := suite.T()
+
+	b := Builder{}
+	name := "itemName"
+	info := ItemInfo{
+		OneDrive: &OneDriveInfo{
+			ItemType: OneDriveItem,
+			ItemName: name,
+		},
+	}
+
+	itemPath := makeItemPath(
+		t,
+		path.OneDriveService,
+		path.FilesCategory,
+		"a-tenant",
+		"a-user",
+		[]string{
+			"drive-id",
+			"root:",
+			"folder",
+			name + "-id",
+		},
+	)
+
+	otherItemPath := makeItemPath(
+		t,
+		path.OneDriveService,
+		path.FilesCategory,
+		"a-tenant",
+		"a-user",
+		[]string{
+			"drive-id",
+			"root:",
+			"folder",
+			name + "-id",
+			name,
+		},
+	)
+
+	require.NoError(t, b.Add(
+		itemPath.String(),
+		"deadbeef",
+		itemPath.ToBuilder().Dir().String(),
+		itemPath.String(),
+		false,
+		info,
+	))
```

Comment:
To prep for the upcoming change:
```suggestion
        err :=  b.Add(
		itemPath.String(),
		"deadbeef",
		itemPath.ToBuilder().Dir().String(),
		itemPath.String(),
		false,
		info)
	require.NoError(t, err)
```